fn_jira/fn_jira/components/jira_common.py

```diff
@@ -0,0 +1,208 @@
+    Return the raw JSON returned from the api call
+    """
+
+    issue_url = '/'.join((appDict['url'], ISSUE_URL))
+
+    payload = _mkCreatePayload(appDict)
+    log.info(payload)
+
+    result = execute_call(log, 'post', issue_url, appDict['user'], appDict['password'], payload)
+    # log.info(result)
+
+    return result
+
+
+def transition_issue(log, appDict):
+    """Function: transition a jira issue.
+    Return the raw JSON returned from the api call
+    """
+
+    url = '/'.join((appDict['url'], TRANSITION_PARAM))
+    payload = _mkTransitionPayload(appDict)
+    log.info(payload)
+
+    result = execute_call(log, 'post', url, appDict['user'], appDict['password'], payload)
+    # log.info(result)
+
+    if appDict['comment']:
+        create_comment(log, appDict)
+
+    return result
+
+def create_comment(log, appDict):
+    """Function: create a jira comment in a Jira issue. No JSON is returned on success"""
+
+    url = '/'.join((appDict['url'], COMMENT_PARAM))
+
+    payload = _mkCommentPayload(appDict)
+    log.info(payload)
+
+    resp =  execute_call(log, 'post', url, appDict['user'], appDict['password'], payload)
+
+    return resp
+
+
+def execute_call(log, verb, url, user, password, payload):
+    """Function: perform the http API call. Different types of http operations are supported: 
+        GET, POST, PUT
+        Errors raise FunctionError
+    """
+    
+    try:
+        headers = {'content-type': 'application/json'}
+
+        log.info(payload)
+
+        if verb == 'get':
+            resp = requests.get(url, verify=False, headers=headers, data=payload, auth=(user, password)) ## verify=False for debugging
+        elif verb == 'post':
+            resp = requests.post(url, verify=False, headers=headers, data=payload, auth=(user, password)) ## verify=False for debugging
+        elif verb == 'put':
+            resp = requests.put(url, verify=False, headers=headers, data=payload, auth=(user, password)) ## verify=False for debugging
+        else:
+            raise FunctionError("unknown verb {}".format(verb))
+
+        if resp is None:
+            raise FunctionError('no response returned')
+
+        if resp.status_code >= 300:
+            log.info(resp)
+            # get the result
+            raise FunctionError(resp.text)
+
+
+        # check if anything returned
+        log.info(resp.text)
+        if resp.text is None or len(resp.text) == 0:
+            return { }          # make sure to always return a dictionary
+
+        # get the result
+        r = resp.json()
+
+        # Produce a FunctionError with the return value
+        return r      # json object needed, not a string representation
+    except Exception as err:
+        raise FunctionError(err)
+
+def _mkCreatePayload(appDict):
+    '''
+    Build the payload for creating a Jira issue
+    :param **dict could be **kwargs:
+    :return: json payload for jira update
+    '''
+    
+    payload = {
+        "fields": {
```

Comment:
We made a decision before about " and ', in order to keep consistency. We use " for string (more than one char) and ' for single char. Not sure if we still want this.